src/ledger.cc

```diff
@@ -168,23 +168,34 @@ void Ledger::process() {
           ++threshold_itr) {
         double pct =
           (*threshold_itr < 0) ? -(*threshold_itr) : (*threshold_itr);
+         int num_in_threshold = 0;
 
         if (count > 1) {
-          double num_in_threshold = round(pct / 100 * count);
+          num_in_threshold = round(pct / 100 * count);
           if (0 == num_in_threshold) {
             continue;
           }
           threshold_boundary = values[num_in_threshold - 1];
         }
 
+        double pct_sum = 0;
+        for (int i = 0; i < num_in_threshold; ++i) {
+            pct_sum += values[i];
+        }
+        double pct_mean = pct_sum / num_in_threshold;
+
         // generate pct name
         char clean_pct[17] = {0};
-        snprintf(clean_pct, sizeof(clean_pct), "upper_%g", pct);
+        snprintf(clean_pct, sizeof(clean_pct), "%g", pct);
```

Comment:
I recall what this was for.  This takes the value of pct and injects it into the variable clean_pct with formatting before it get converted to a display value.  So this is necessary I think.